packages/api/src/cors.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-env serviceworker */
+import type { RouteHandler } from 'itty-router'
+
+/**
+ * Handler for CORS OPTIONS requests.
+ */
+export function corsOptions (request: Request): Response {
+  const headers = request.headers
+  // Make sure the necessary headers are present for this to be a valid pre-flight request
+  if (
+    headers.get('Origin') !== null &&
+    headers.get('Access-Control-Request-Method') !== null &&
+    headers.get('Access-Control-Request-Headers') !== null
+  ) {
+    function headerOrDefault (key: string, fallback: string): string {
+      const value = headers.get(key)
+      if (value !== '' && value !== null) {
+        return value
+      }
+      return fallback
+    }
+    // Handle CORS pre-flight request.
+    /** @type {Record<string, string>} */
+    const respHeaders = {
+      'Content-Length': '0',
+      'Access-Control-Allow-Origin': headerOrDefault('origin', '*'),
+      'Access-Control-Allow-Methods': 'GET,POST,DELETE,OPTIONS',
+      'Access-Control-Max-Age': '86400',
+      // Allow all future content Request headers to go back to browser
+      // such as Authorization (Bearer) or X-Client-Name-Version
+      'Access-Control-Allow-Headers':
+        headerOrDefault('Access-Control-Request-Headers', ''),
+      'Access-Control-Expose-Headers': 'Link, Count, Page, Size'
```

Comment:
I don't think we need these as `Link, Count, Page, Size'` are all for pagination which w3name doesn't use

Reply:
Good point.